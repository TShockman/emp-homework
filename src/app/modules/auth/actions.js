export const NAME = 'auth';

// lifecycle action names for a login request
export const LOGIN_REQUESTED = `${NAME}/LOGIN_REQUESTED`;
export const LOGIN_FULFILLED = `${NAME}/LOGIN_FULFILLED`;
export const LOGIN_REJECTED = `${NAME}/LOGIN_REJECTED`;
// NOTE: currently the rejected action is unused for simplicity sake
// since logging in always succeeds anyways. In a fuller implementation,
// LOGIN_REJECTED would be dispatched if the service request failed, the
// auth reducer would update the auth state to signify this, and the Login
// component would handle displaying relevant error text to the user.