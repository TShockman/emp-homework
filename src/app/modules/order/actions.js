export const NAME = 'order';

// action names for get orders flow
export const GET_ORDERS_REQUESTED = `${NAME}/GET_ORDERS_REQUESTED`;
export const GET_ORDERS_FULFILLED = `${NAME}/GET_ORDERS_FULFILLED`;
