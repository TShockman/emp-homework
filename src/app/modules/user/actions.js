export const NAME = 'user';

export const USER_REQUESTED = `${NAME}/USER_REQUESTED`;
export const USER_FULFILLED = `${NAME}/USER_FULFILLED`;
