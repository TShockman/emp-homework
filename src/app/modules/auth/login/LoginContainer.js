import {connect} from 'react-redux'
import {LOGIN_REQUESTED} from '../actions';
import Login from './Login';
import {selectAuthState} from '../selectors';


function mapStateToProps(state) {
  const {id} = selectAuthState(state);
  return {
    id
  };
}

function mapDispatchToProps(dispatch) {
  return {
    requestLogin: ({username, password}) => dispatch({
      type: LOGIN_REQUESTED,
      username,
      password
    })
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Login);