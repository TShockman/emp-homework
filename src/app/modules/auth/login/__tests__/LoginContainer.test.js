/* eslint-disable import/first */
jest.mock('../Login', () => 'Login');

import React from 'react';
import renderer from 'react-test-renderer';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import authReducer, {AuthState} from '../../reducer';
import {NAME} from '../../actions';
import LoginContainer from '../index';

describe('LoginContainer', () => {
  it('Provides Login with correct props', () => {
    const store = createStore(authReducer, {
      [NAME]: new AuthState({id: 123})
    });
    const tree = renderer.create(
        <Provider store={store}>
          <LoginContainer/>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
