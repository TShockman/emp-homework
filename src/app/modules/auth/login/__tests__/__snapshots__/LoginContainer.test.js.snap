// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginContainer Provides Login with correct props 1`] = `
<Login
  id={123}
  requestLogin={[Function]}
/>
`;
